### PR TITLE
Propagate `maxIter` parameter and enable `prove_loop`

### PR DESCRIPTION
**Pull request by zoep**
This is a fix for #100. Summary of changes:
- Propagate the `UnitTestOptions` to VeriOpts in SymExec.hs.
- Add debug option in `UnitTestOptions`
- Move `prove_loop` in a separate file
- Run `prove_loop` with two different `maxIter` params (passing and failing)

### CODE REVIEW

File: src/hevm/src/EVM/UnitTest.hs [hunk 1]
```diff
@@ -74,6 +75,7 @@ data UnitTestOptions = UnitTestOptions
   , verbose     :: Maybe Int
   , maxIter     :: Maybe Integer
   , askSmtIters :: Maybe Integer
+  , debug       :: Bool
```
Comment (d-xo) [inline src/hevm/src/EVM/UnitTest.hs:78]: could we call this smtdebug? just to make the distinction between dumping the smt queries and running the visual debugger clear?

**Review by d-xo — COMMENTED** [review]
Looks great! just a little nit re: naming

File: src/hevm/src/EVM/UnitTest.hs (hunk 1 above)
Comment (zoep) [reply src/hevm/src/EVM/UnitTest.hs:78]: Sure! Do you think we should change the `bebug` option in `VeriOpts` as well or only this one?

Comment (zoep) [reply src/hevm/src/EVM/UnitTest.hs]: Left `VeriOpts` as is since it prints nonSMT stuff as well

**Pull request merged**